---
title: "MSY subunits overlapping ChIP-seq anti-DMC1 peaks"
author: "Lucas A. Nell"
date: "March 25, 2016"
output: 
html_document:
    self_contained: true
    highlight: haddock
runtime: shiny
---


```{r setup, include = FALSE, eval = TRUE, echo = FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = TRUE, comment = '', prompt = FALSE)
knitr::read_chunk('ChIP_subunits.R')
```

# Preamble

## Required packages
```{r packages, message = FALSE}
```

## `ggplot2` theme and palettes
A minimalistic theme for plots, plus fill and color palettes.
```{r ggplot_theme}
```



# Table of subunit locations on mouse Y-chromosome long arm

## Download
I first downloaded Table S4 (`'mmc4.xlsx'`) from Soh et al. (2014) in *Cell* 
([link](http://dx.doi.org/10.1016/j.cell.2014.09.052)).

I'll assume you've downloaded it to directory `table_dir`, which for me was simply `'.'`.

## Initial read
The initial excel file included column names too long for my tastes, so I skipped the
header row and made my own column names. I also specified column types for reasons of
paranoia.
I then saved this data frame as a compressed `.RData` file, which is much smaller than a 
non-compressed `.xlsx`.

```{r initial_read, eval = FALSE}
table_dir <- '.'

subunit_df <- read_excel(paste0(table_dir, '/mmc4.xlsx'),
                       skip = 1, 
                       col_names = c('start', 'end', 'subunit'),
                       col_types = c('numeric', 'numeric', 'text'))
save(subunit_df, file = 'ChIP_subunits.RData', compress = TRUE)
```

## Subsequent reads
After the initial processing, I could remove the `mmc4.xlsx` file and simply `load` the
`RData` file made previously.

```{r load_RData}
```



# ChIP-seq peak files

I next input the `'*.bsites'` files that were output from the SISSRs peak-finding 
algorithm. I wanted to combine tables from multiple samples, plus filter for only 
peaks located in the male-specific Y (MSY) long arm. In the filename for the `*.bsites`
file, the suffix `"_Y_Dq_Pv01"` indicates it's aligned to chromosome Y, used the 
anti-DMC1 antibody, was filtered for MAPQ ≥ 20, and had a cutoff p-value of 0.01.

> *Note*: Here, sample name and strain name are synonymous since I only have one sample
> per strain. If this changes, obviously this code will need to be updated, too.


```{r input_peaks}
```



# Preliminary comparison of subunits to ChIP-seq peak densities

The following plot shows how peak starting locations (bottom rug plot) relate to the
density of subunit starting locations (histogram) along the MSY long tail.

```{r comparison_plot}
```



# Counting subunits

The function below is to find the number of subunits within a given sized 
'flank' on each side of peak's center. I chose a flank of 10 Mb (for a total window size
= 2 Mb + 1 b). I used `sapply` so that I can input a vector for `centers`, 
instead of just one position at a time. This makes computations faster (vs for-loop), 
and makes future code much simpler.

```{r over_counts}
```


I next created new columns with peak center positions, and the # of overlapping subunits.
I grouped by sample because I don't want to aggregate all calculations over all samples. 
I lastly made the `sample` column into a factor for easy plotting, and made a summary 
data frame for later calculations.

```{r compute_overs}
```





# Bootstrap peak centers


## Set range of possible peak centers
I set the range of possible center locations based on two facts:

1. The length of narrowest ChIP-seq peak should be 41 based on SISSRs algorithm
2. Widest possible range of values for centers (since I filtered `peak_df` for 
whether it's within the MSY based on start and end positions, not centers) should be 
from `MSY_start + narrowest_center` to `MSY_end - narrowest_center`


```{r bootInterval, eval = FALSE}
narrowest_center <- median(c(1, 41))
possible_centers <- seq(min(subunit_df$start) + narrowest_center, 
                     max(subunit_df$end) - narrowest_center)
```


## Run bootstraps
Function `one_boot` runs a single bootstrap replicate, given a sample size `n`, by 
sampling `n` locations within `possible_centers` (with replacement) and counting the 
number of overlapping subunits for each. 

I ran 10,000 bootstrap resamples, of sample size n~1~ + n~2~ + n~3~ (for all 3
samples from `peak_df`), and output results to `cent_boot_matrix`.

```{r bootstrapping, eval = FALSE}
one_boot <- function(n){
    boot_centers <- sample(possible_centers, replace = TRUE, size = n)
    boot_over_counts <- over_counts(boot_centers)
    return(boot_over_counts)
}

set.seed(80502)
cent_boot_matrix <- replicate(1e4, expr = one_boot(n = nrow(peak_df)))
```

```{r input_boot_results, echo = FALSE}
```




## Organize bootstrap output matrix

The output from bootstrapping, `cent_boot_matrix`, is in a pretty ugly format: a 75 row,
10,000 column matrix.

```{r boot_out_head}
str(cent_boot_matrix)
```


### Convert to tibble data frame

To make this much easier to work with, I'm going to first convert the matrix to a 
`tbl_df` (pronounced "tibble data frame"), which is a data frame with better 
printing, among other attributes. 
(I use these types of data frames almost exclusively, but this is the only portion of 
code where the difference is necessary to explain.)
For more information on tibble data frames, run `vignette('tibble', 'tibble')`

The package `tibble` is already installed with `dplyr`, but `dplyr` has its own
`as_data_frame` command that we don't want to use here. 
This is why I specify the package by using `tibble::as_data_frame`.

The new column `sample` I create is directly from the `peak_df` data frame and will be 
used later to split the bootstrap resamples by sample.

The `gather` function converts data from "wide" format to "tall" (i.e., fewer columns, 
more rows). See [this link](https://rpubs.com/bradleyboehmke/data_wrangling) for 
more information.

The `gather` command made the `num` column contain the column names 
`"V1" ... "V10000"`. I changed that to an integer from 1 to 10,000 indicating the 
bootstrap number.

```{r boot_to_df}
```


### Summarize by bootstrap

I want to compare overlapping subunit mean and variability between observed and 
bootstrap resamples, so I need to summarize our bootstrapping results. 
I first summarize by sample and bootstrap number, calculating the mean and standard 
deviation for each sample, for each bootstrap. Our final `cent_boot_df` data frame
contains 10,000 rows (1 for each bootstrap) for each sample, so 30,000 rows total.

```{r boot_summ_by_bootstrap}
```


### Summarize by sample

I now construct a p-value data frame by grouping by sample, then assessing, for each 
sample, the proportion of bootstrapped means were ≥ the observed mean and standard
deviations were ≤ the observed standard deviation.

```{r boot_summ_by_sample}
```






# Plots


## Mean overlapping subunits

Histogram of the mean numbers of subunits within flanks for 10^4^ bootstrap resamples of
positions along MSY long arm, in relation to mean subunits in flanks for each sample 
(dashed lines). P-values for tests of H~0~: µ~sample~ ≤ µ~random~ are shown.


```{r boot_plot_mean}
```



## Variability in overlapping subunits

Histogram of the standard deviation of subunits within flanks for 10^4^ bootstrap 
resamples of positions along MSY long arm, in relation to standard deviations of 
subunits in flanks for each sample (dashed lines). P-values for tests of H~0~: 
σ~sample~ ≥ σ~random~ are shown.


```{r boot_plot_sd}
```


<!-- rmarkdown::render('ChIP_subunits.Rmd') -->
